openslides_backend/action/actions/meeting_user/set_data.py

```diff
@@ -30,8 +32,8 @@ class MeetingUserSetData(UpdateAction):
     )
 
     def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
My approach:
```
meeting_id = instance.pop("meeting_id", None)
user_id = instance.pop("user_id", None)
if instance.get("id"):
   fqid = ....
   meeting_user = self.datastore.get(fqid, [], raise_exception=True)
   if meeting_id:
       assert meeting_id == meeting_user["meeting_id"], "Not permitted to change ..."
   if user_id:
       assert user_id == meeting_user["user_id"], "Not permitted to change ..."
elif meeting_id and user_id:
        meeting_users = self.datastore.filter(
            "meeting_user",
            And(
                FilterOperator("meeting_id", "=", meeting_id),
                FilterOperator("user_id", "=", user_id),
            ),
            ["id"],
        ).values()
        if not meeting_users:
            ... continue with existing code
else:
   raise exception either id nor meeting/user_ids
```

Reply:
It looks to complicated. I prefer my solution.

Reply:
I am working on it. I will fill the blanks.

Reply:
The check_archived_meeting code is called before the update_instance code. So I need to throw the exception in the get_meeting_id method. So I overwrite it.